sdk/debugger_interface/linux/gdb-sgx-plugin/gdb_sgx_plugin.py

```diff
@@ -686,6 +697,46 @@ def stop(self):
             gdb.execute(gdb_cmd, False, True)
         return False
 
+# This breakpoint is to handle enclave creation with buffer
+class CreateBufferEnclaveBreakpoint(gdb.Breakpoint):
+    def __init__(self):
+        gdb.Breakpoint.__init__ (self, spec="_create_enclave_from_buffer_ex", internal=1)
+
+    def stop(self):
+        bp_in_urts = is_bp_in_urts()
+        if bp_in_urts == True:
+            # Get se_file_t pointer (4th parameter)
+            file_addr = gdb.parse_and_eval("$rcx")
+            file_str = read_from_memory(file_addr, 8 + 2*4)
+            file_tuple = struct.unpack_from("QII", file_str)
+
+            if file_tuple[1] == 0:
+                # If it is null, then it does not have a file. So we dump the buffer
+
+                # dump enclave to file
+                dump_name = "/tmp/enclave_dump_{0}.bin".format(time.time())
+                buffer_ptr = gdb.parse_and_eval("$rsi")
```

Comment:
The code assumes we are under 64-bit. Currently we still need to support 32-bit so we may need to add that support as well just like we do for several other internal breakpoints

Reply:
Hey, thank you for your review. Indeed I was thinking the same thing about the support for 32-bit as the SDK and SGX is also support 32-bit system. However, I am not sure how to test that since I am running the 64-bit system. I assume I will need to do either cross-compiling or install a 32-bit VM. Perhaps the difference is that the debugger now needs to inspect the stack as the function parameter is now spilled to the stack instead of register. But I guess we can do that. I'll see if I can help in the coming days :)